src/launcher/java/org/truffleruby/launcher/RubyLauncher.java

```diff
@@ -76,11 +76,18 @@ protected void printVersion() {
             }
 
             // Put java options to polyglot options
-            for (Map.Entry<String, String> jvmOptionEntry : config.getJVMOptions().entrySet()) {
-                polyglotOptions.put(
-                                // ignore dash at the beginning
-                                "jvm." + jvmOptionEntry.getKey().substring(1),
-                                jvmOptionEntry.getValue());
+            for (String jvmOption : config.getJVMOptions()) {
+                final int eqIdx = jvmOption.indexOf('=');
+                final String key;
+                final String value;
+                if (eqIdx < 0) {
+                    key = jvmOption.substring(2);
```

Comment:
What's this substring(2)?
It seems common to both branches, so it seems it could be done before the condition.

Reply:
Doesn't matter since this is removed in later commits.